tools/android/packaging/xbmc/src/org/xbmc/kodi/XBMCVideoView.java.in

```diff
@@ -17,83 +11,86 @@ import android.content.Context;
 import android.graphics.Canvas;
 import android.graphics.Color;
 import android.view.View;
+import android.widget.RelativeLayout;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.FutureTask;
 
 public class XBMCVideoView extends SurfaceView implements
     SurfaceHolder.Callback
 {
+  native void _OnSurfaceChanged(SurfaceHolder holder, int format, int width, int height);
+  native void _OnSurfaceCreated(SurfaceHolder holder);
+  native void _OnSurfaceDestroyed(SurfaceHolder holder);
+
   private static final String TAG = "XBMCVideoPlayView";
 
-  public boolean mHasHolder = false;
-  public int mWidth = -1;
-  public int mHeight = -1;
+  public boolean mIsCreated = false;
+  private RelativeLayout mVideoLayout = null;
 
-  public XBMCVideoView(Context context)
+  public static XBMCVideoView createVideoView()
   {
-    super(context);
-    getHolder().addCallback(this);
+    FutureTask<XBMCVideoView> futureResult = new FutureTask<XBMCVideoView>(new Callable<XBMCVideoView>() {
+      @Override
+      public XBMCVideoView call() throws Exception
+      {
+        return new XBMCVideoView(Main.MainActivity);
+      }
+    });
+
+    try
+    {
+      Main.MainActivity.runOnUiThread(futureResult);
+      XBMCVideoView vw = futureResult.get();
```

Comment:
As such, surely, but this java construct is quite abstract, so let's make it obvious what is returned, here.